pyanaconda/installation.py

```diff
@@ -234,22 +234,15 @@ def _prepare_installation(payload, ksdata):
     installation_queue.task_completed.connect(lambda x: progress_step(x.name))
 
     # This should be the only thread running, wait for the others to finish if not.
+    #
+    # We need to do this before we start filling the task queue, or else the DBus
+    # tasks we create might have stale data due to background processing threads
+    # still running at task creation time.
     if threadMgr.running > 1:
-        # it could be that the threads finish execution before the task is executed,
-        # but that should not cause any issues
-
-        def wait_for_all_treads():
-            for message in ("Thread %s is running" % n for n in threadMgr.names):
-                log.debug(message)
+        for message in ("Thread %s is running" % n for n in threadMgr.names):
+            log.debug(message)
             threadMgr.wait_all()
-
-        # Use a queue with a single task as only TaskQueues have the status_message
-        # property used for setting the progress status in the UI.
-        wait_for_threads = TaskQueue("Wait for threads to finish",
-                                     N_("Waiting for %s threads to finish") % (threadMgr.running - 1))
```

Comment:
I think we should call `progressQ.send_message` before the waiting to show the same message on the progress screen as before.

Reply:
Good point, added - as far as I can tell, the output now looks similar to what was shown before, possibly just in different order due to the insufficient synchronization before.